### DISCUSSION TITLE
Histogram panel should handle both classic and native histograms correctly for Prometheus datasource

### DISCUSSION THREAD

**Discussion by krajorama** [grafana/grafana#69789] [Dashboards, Panels & Visualizations]
Histogram panel can display "classic" histograms from the Prometheus data source. The classic histograms were encoded as several time series representing cumulative buckets and a count and sum series.

Prometheus has introduced [native histograms](https://prometheus.io/docs/prometheus/latest/feature_flags/#native-histograms). This encodes sparse/exponential high resolution histograms into a single time series. This means the time series no longer contains floating point numbers, rather a complex data structure.

Result format on data source: https://prometheus.io/docs/prometheus/latest/querying/api/#native-histograms

For more information see presentations at PromCon for example: https://www.youtube.com/watch?v=RDdlDRkuVL8&t=2780s

![image (6)](https://github.com/grafana/grafana/assets/13435893/438a5632-e137-44cf-92e9-312246822d7c)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This discussion has an issue - #69790. Feel free to continue the discussion around this there. Thank you!
